aptos-move/aptos-vm-types/src/abstract_write_op.rs

```diff
@@ -193,6 +197,7 @@ pub struct WriteWithDelayedFieldsOp {
 pub struct InPlaceDelayedFieldChangeOp {
     pub layout: Arc<MoveTypeLayout>,
     pub materialized_size: u64,
+    pub metadata: StateValueMetadata,
```

Comment:
Sorry I missed this change - why do InPlaceDelayedFieldOps now carry metadata? @msmouse @igor-aptos

Reply:
if the size changes, we need to change the deposit

Reply:
you mean multiple ops that get squashed? 

each session starts from the output of the previous, so:

we start with "storage_size", and that is deposit in the metadata

then first session modifies size to "size_1", we charge for "size_1 - storage_size", and metadata is updated to size_1

then second session modifies to "size_2", we charge for "size_2 - size_1", and metadata is updated to size_2


squashing just needs to take the most recent metadata?

Reply:
sorry, I was confused about the struct, mis-remembered how it's used.
as per our discussion though, we should probably make the handling uniform now with at least ResourceGroupInPlaceDelayedFieldChangeOp and possibly even GroupWrites (and maybe avoid change to modification thingy we are doing there)